gui/system/views.py

```diff
@@ -1975,19 +1976,99 @@ def CA_export_privatekey(request, id):
     return response
 
 
+def certificate_progress(request):
+    with client as c:
+        job = c.call(
+            'core.get_jobs',
+            [['id', '=', request.session['certificate_create']['job_id']]],
+            {'get': True}
+        )
+
+    return HttpResponse(json.dumps({
+        "status": "finished" if job["state"] in ["SUCCESS", "FAILED", "ABORTED"] else job["progress"]["description"],
+        "volume": job["arguments"][0],
+        "extra": job["progress"]["extra"],
+        "percent": job["progress"]["percent"],
+    }), content_type='application/json')
+
+
+def certificate_acme_create(request, csr_id):
+    if request.session.get('certificate_create'):
+
+        form, job, verrors = certificate_common_post_create(
+            'create', request, csr_id=csr_id
+        )
+
+        if job['state'] == 'SUCCESS':
+            return JsonResp(request, message=_('ACME Certificate successfully created'))
+        else:
+            if not verrors:
+                raise ClientException(job['error'], trace=job['exception'])
+            else:
+                return render(request, 'system/certificate/certificate_create_ACME.html', {
+                    'form': form
+                })
+
+    if request.method == "POST":
+        form = forms.CertificateACMEForm(request.POST, csr_id=csr_id)
+        if form.is_valid():
+            job_id = form.save()
+            request.session['certificate_create'] = {
+                'job_id': job_id,
+                'form': form.__class__.__name__,
+                'payload': request.POST,
+            }
+
+            return render(
+                request, 'system/certificate/certificate_progress.html', {
+                    'certificate_url': reverse('certificate_acme_create', kwargs={'csr_id': csr_id}),
+                    'dialog_name': 'Create ACME Certificate'
+                }
+            )
+        return JsonResp(request, form=form)
+
+    else:
+        form = forms.CertificateACMEForm(csr_id=csr_id)
+
+    return render(request, "system/certificate/certificate_create_ACME.html", {
+        'form': form
+    })
+
+
 def certificate_csr_import(request):
 
+    if request.session.get('certificate_create'):
+
+        form, job, verrors = certificate_common_post_create(
+            'create', request
+        )
+
+        if job['state'] == 'SUCCESS':
+            return JsonResp(request, message=_('Certificate Signing Request successfully Imported'))
+        else:
+            if not verrors:
+                raise ClientException(job['error'], trace=job['exception'])
```

Comment:
Legacy UI should raise MiddlewareError, ClientException would not make sense in this context I would think?